ui/src/app/controllers/components/ControllerHeaderForms/AddController/AddController.test.tsx

```diff
@@ -0,0 +1,58 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import AddController from "./AddController";
+
+import type { RootState } from "app/store/root/types";
+import {
+  configState as configStateFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("AddController", () => {
+  let state: RootState;
+
+  beforeEach(() => {
+    state = rootStateFactory({
+      config: configStateFactory({
+        items: [
+          { name: "maas_url", value: "http://1.2.3.4/MAAS" },
+          { name: "rpc_shared_secret", value: "veryverysecret" },
+        ],
+      }),
+    });
+  });
+
+  it("includes the config in the instructions", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter>
+          <AddController clearHeaderContent={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    const instructions = wrapper.find("[data-testid='instructions']").text();
+    expect(instructions.includes("http://1.2.3.4/MAAS")).toBe(true);
+    expect(instructions.includes("veryverysecret")).toBe(true);
+  });
+
+  it("can close the instructions", () => {
+    state.zone.loaded = false;
+    const store = mockStore(state);
+    const clearHeaderContent = jest.fn();
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter>
+          <AddController clearHeaderContent={clearHeaderContent} />
+        </MemoryRouter>
+      </Provider>
+    );
+    wrapper.find("Button[data-testid='close']").simulate("click");
```

Comment:
Doesn't using a generic string ("close") defeat the purpose of using a testid?

This could be re-written with react testing library: `userEvent("click", getByRole("button", { name: "Close" }))`. It would the advantage of testing actual DOM nodes that users interact with instead of enzyme wrappers.

Reply:
Thanks, yep I suppose one of the child components could also include that generic string. The RTL example you've provided wouldn't make a conflict any less likely that a child component might provide a conflicting element.

I've added a prefix to the test id.